### PR TITLE
chore(auto-update-manager): add logging for autoupdates

### PR DESCRIPTION
**Pull request by mcasimir**
Cover auto-update with logging

NOTE: this also changes the initialization sequence slightly so that logging and telemetry are set up before the rest

### CODE REVIEW

File: packages/compass/src/main/application.ts [hunk 1]
```diff
@@ -44,7 +40,7 @@ class CompassApplication {
   }
 
   static init(): Promise<void> {
-    return this.initPromise ??= this._init();
+    return (this.initPromise ??= this._init());
```
Comment (mcasimir) [inline packages/compass/src/main/application.ts:43]: i don't know what did this and why, I'm assuming the kind of eslint / prettier

File: packages/compass/src/main/auto-update-manager.ts [hunk 2]
```diff
@@ -1,55 +1,85 @@
...
-const API_PLATFORM = {
+const API_PLATFORM: Record<string, string> = {
   darwin: 'osx',
   win32: 'windows',
   linux: 'linux',
 };
 
-function isSupportedPlatform(str?: string): str is keyof typeof API_PLATFORM {
-  return Object.keys(API_PLATFORM).includes(str as string);
-}
```
Comment (gribnoysup) [inline packages/compass/src/main/auto-update-manager.ts:33]: Any particular reason we removed this? Don't have a strong preference, but I like how explicit it was

**Review by gribnoysup — APPROVED** [review]

File: packages/compass/src/main/auto-update-manager.ts (hunk 2 above)
Comment (mcasimir) [reply packages/compass/src/main/auto-update-manager.ts:33]: Absolutely nothing wrong with that code from my point of view, but types were complaining in VSCode, which is weird. It is still understandable what's going on cause the logs I added basically explain it. We can also take a look at the warnings i was getting if you want

Comment (gribnoysup) [reply packages/compass/src/main/auto-update-manager.ts:33]: Totally can do this separately. I'd be happy to figure out what's up with VSCode, but don't have time right away and don't want to block this PR

**Pull request merged**